stac_generator/point_stac_generator.py

```diff
@@ -0,0 +1,353 @@
+import datetime as pydatetime
+from collections.abc import Sequence
+from itertools import chain
+from pathlib import Path
+from typing import Any, Literal, cast
+from uuid import UUID, uuid4
+
+import numpy as np
+import pandas as pd
+import pystac
+from pystac.extensions.projection import ItemProjectionExtension
+
+from stac_generator.generator import StacGenerator
+from stac_generator.typing import (
+    BBoxT,
+    CSVMediaType,
+    DateTimeT,
+    FrameT,
+    GeometryObj,
+    PointBandInfo,
+    TimeExtentT,
+)
+
+
+# Helper functions
+def ensure_name_suffix(name: str, suffix: str) -> str:
+    """Ensure that name ends with suffix.
+    If name does not end with suffix, add suffix to the end of name
+
+    Args:
+        name (str): name
+        suffix (str): suffix value
+
+    Returns:
+        str: name with suffix
+    """
+    if name.lower().endswith(suffix):
+        return name
+    if name.endswith("_"):
+        return name + suffix
+    return name + "_" + suffix
+
+
+def create_item_name(collection_name: str, group_name: str | None = None) -> str:
+    suffix = group_name + "_item" if group_name else "item"
+    # Collection name is guaranteed to contain collection suffix
+    assert collection_name.endswith("collection")
+    parts = collection_name.rsplit("collection", 1)
+    parts[-1] = suffix
+    return "".join(parts)
+
+
+def get_path(path: str | Path) -> Path:
+    if isinstance(path, Path):
+        return path
+    return Path(path)
+
+
+def calculate_spatial_extent(
```

Comment:
Like you said this can probably be solved with geopandas. I think its a good idea to use geopandas instead of regular pandas whenever there is a spatial component to the data. 

```
import geopandas as gpd

gdf = gpd.read_file(filepath)
print(gdf.geometry.total_bounds())

> array([min_X, min_Y, max_X, max_Y])
```

Reply:
Resolved in the latest commit